scripts/index/html/artifacts.js

```diff
@@ -0,0 +1,354 @@
+                " in scheme 'releases/<type>/<name>/<platform>/<arch>/...'" +
+                " for path '" + path + "'");
+      }
+
+      self.arch = parts[4];
+
+      if (self.type === "binary") {
+          var regex = /(\d+)\.(\d+).(\d+)/;
+          matches = regex.exec(parts[5]);
+
+          if (matches === null) {
+              throw("Invalid path: The binary version '" + parts[5] + "'" +
+                    " does not match '<major>.<minor>.<patch>' as all" +
+                    " numerical digits in scheme " +
+                    " 'releases/<type>/<name>/<platform>/<arch>/<version>/...'" +
+                    " for path '" + path + "'");
+          }
+
+          self.version = new BinaryVersion(matches[1], matches[2], matches[3]);
+
+          if (parts[6] !== self.name && parts[6] !== self.name + ".exe") {
+              throw("Invalid path: The binary name '" + parts[6] + "'" +
+                    " does not match '<name>{.exe}' in scheme" +
+                    " 'releases/<type>/<name>/<platform>/<arch>/<version>/<name>{.exe}'" +
+                    " for path '" + path + "'");
+          }
+      }
+
+      if (self.type == "plugin") {
+          var regex = /(.+)-(\d+\.\d+)-patch\.(\d+)\.zip/;
+          matches = regex.exec(parts[5]);
+
+          if (matches === null || matches[1] !== self.name) {
+              throw("Invalid path: The plugin name '" + parts[5] + "'" +
+                    " does not match" +
+                    " '<name>-<dcos-version>-patch.<patch-version>.zip' in scheme" +
+                    " 'releases/<type>/<name>/<platform>/<arch>/<version>/<name>-<dcos-version>-patch.<patch-version>.zip'" +
+                    " for path '" + path + "'");
+          }
+
+          self.version = new PluginVersion(matches[2], matches[3]);
+      }
+    };
+
+    // The 'anchor' function constructs an HTML anchor to
+    // an artifact's path, using 'text' as the placeholder
+    // text. If 'text' is 'null', the path itself is used as
+    // the placeholder text.
+    this.anchor = function(text=null) {
+      if (text === null) {
+        text = this.path
+      }
+      return '<a href="' + this.path + '">' + text + '</a>';
+    }
+
+    constructor(this, path);
+  };
+
+  // The 'Artifacts' object defines a collection of
+  // functions for working with a set 'Artifact' objects.
+  this.Artifacts = function(artifacts) {
+    this.artifacts = artifacts;
+
+    // The 'Binaries' object defines a collection of functions
+    // for working with a set of binary 'Artifact' objects.
+    var Binaries = function(binaries) {
+      this.binaries = binaries;
+
+      // List all binary artifacts.
+      this.list = function() {
+        return this.binaries;
+      };
+
+      // List unique major versions across all binaries.
+      this.majorVersions = function() {
+        var versions = [];
+        $.each(this.binaries, function(_, binary) {
```

Comment:
All occurrences of `$.each(a, function(b, c) {` should be replaceable by `a.forEach(function(c, b){`

Reply:
Is there a preference? I've often seen $.each() used all over the place in favor of forEach, but I'm not sure why. Can you explain and / or provide a link?

Reply:
It is a preference, for this index jQuery could be dropped as you can see in http://youmightnotneedjquery.com/ (nearly all of the jQuery helper functions now exist in vanilla JS). As it would require refactoring and the JS code of this PR is clean, I will resolve this conversation and I don't think that changing any `$` calls is necessary.